middlewares/passport-client-cert.js

```diff
@@ -0,0 +1,112 @@
+'use strict';
+
+const util = require('util')
+const Strategy = require('passport-strategy')
+
+/**
+ * `Strategy` constructor.
+ *
+ * The client cert authentication strategy authenticates requests based on the
+ * client certificate credentials submitted in the TLS handshake
+ *
+ * Applications must supply a `verify` callback which accepts a `username` and
+ * an `info` object, which includes the certificate `fingerprint`. It then
+ * calls the `done` callback supplying a `user`. User should be set to `false`
+ * if the credentials are not valid.  If an exception occured, `err` should be
+ * set.
+ *
+ * Options:
+ *   - `passReqToCallback`  when `true`, `req` is the first argument to the
+ *      verify callback (default: `false`)
+ *
+ * Examples:
+ *
+ * passport.use(new ClientCertStrategy(
+ *   function (userName, info, done) {
+ *     if (!config.auth.client_certificates_enabled) {
+ *       return done(new UnauthorizedError('Unsupported authentication method'))
+ *     }
+ *
+ *     Account.findByName(userName)
+ *       .then(function (userObj) {
+ *         if (!userObj || userObj.is_disabled || userObj.fingerprint !== fingerprint) {
+ *           return done(new UnauthorizedError('Unknown or invalid account'))
+ *         }
+ *         done(null, userObj)
+ *       })
+ *   }))
+ *
+ *
+ * @param {Object} options
+ * @param {Function} verify
+ * @api public
+ */
+class ClientCertStrategy extends Strategy {
+  constructor (options, verify) {
+    if (typeof options == 'function') {
```

Comment:
We could, but this replicates the behavior of all the other passport middleware.